### ISSUE TITLE
Travis and Coveralls Integration

### ISSUE THREAD

**Issue by ohyde** [ohyde/ezM#5]
Would work fine for initial cut yeah?

[https://travis-ci.org/](https://travis-ci.org/)
[https://coveralls.io/](https://coveralls.io/)

I'll add as todo for client project.

**Comment by lucasschejtman** [comment]
The thing with travis/coveralls is that they only expect one project per repo, so you can add it for the front end stuff but we wouldn't be able to also do it for the back end (unless they've changed the free tier)

Having said that, go for it

**Comment by kahyde** [comment]
Why do they only expect one project? Is that a standard way of organising things, or is it just to encourage people to pay? If it is standard, why is this project organised differently?

**Comment by ohyde** [comment]
These tools that I referenced, in particular, are reasonably coupled to individual GitHub projects, at least to the extent that I've used them. 

The benefit of this is that it's very easy to get up and running with one, normally getting a build and coverage pipeline setup can be somewhat tedious because of all the configuration you have to set up. 

It's not a standard way, you usually have much more control over what get's built, you can even sometimes checkout multiple repositories in the same build pipeline ).

This project is organized differently so that Lucas and I could keep track of what each other is working one easily while the code base spins up, once the individual projects ( server/client ) grow large enough they'll be split and each will be able to use travis/coveralls on their own. 

The server stuff, for example, is actually going to be multiple projects, e.g.

Repositories:
ezMClient ( for examples sake )
invoicesServerless
expensesServerless
usersServerless
etc etc

**Comment by kahyde** [comment]
So would travis be somewhat similar to Jenkins. And coveralls similar to checkstyle but for coverage rather than style?

**Comment by ohyde** [comment]
Yeah, Travis will be similar to Jenkins, TeamCity, Bamboo, Bitbucket Pipelines, Circle CI and a few others available. They all have different selling points, the tendency to lean towards Travis in this instance is that it is very easy to set up and cheap for a small project that may not get off the ground.

Coveralls is slightly different to checkstyle, as far as I can see checkstyle would be more similar to something like 'ESLint' which keeps track of your coding style.

Coveralls is just a pretty reporter for your unit test code coverage. I like it because I like pretty things that make me feel productive.

![image](https://user-images.githubusercontent.com/794829/27201742-e436c22e-5272-11e7-8d67-1f4487e3ed9e.png)

This is a screenshot from one of my other projects that have it setup. This will essentially help us keep track of situations where we add loads of features without tests to go with them, the reports will start to show lesser and lesser % covered.

**Comment by lucasschejtman** [comment]
any excuse to show that 100% coverage
